### PR TITLE
[REF] Permit domain tokens being used within Thank You letters

### PR DESCRIPTION
**Pull request by seamuslee001**
Overview
----------------------------------------
Contribution Thank you letters do not output domain tokens e.g. {domain.name} 

Before
----------------------------------------
Domain tokens are not outputted in Contribution Thank you letters

After
----------------------------------------
Domain tokens are replaced in Contribution Thank You letters

ping @mattwire @eileenmcnaughton @monishdeb

### CODE REVIEW

**Comment by monishdeb** [comment]
Changes looks good to me. 

* General standards
    * Explain ([`r-explain`](https://docs.civicrm.org/dev/en/latest/standards/review/#r-explain)) PASS
    * User impact ([`r-user`](https://docs.civicrm.org/dev/en/latest/standards/review/#r-user)) PASS
    * Documentation ([`r-doc`](https://docs.civicrm.org/dev/en/latest/standards/review/#r-doc)) PASS
    * Run it ([`r-run`](https://docs.civicrm.org/dev/en/latest/standards/review/#r-run)) PASS
* Developer standards
    * Technical impact ([`r-tech`](https://docs.civicrm.org/dev/en/latest/standards/review/#r-tech)) PASS
    * Code quality ([`r-code`](https://docs.civicrm.org/dev/en/latest/standards/review/#r-code)) 
        * __PASS__
           * Test results ([`r-test`](https://docs.civicrm.org/dev/en/latest/standards/review/#r-test))
        * __PASS__: The test results are all-clear.

**Review by monishdeb — APPROVED** [review]

File: CRM/Contribute/Form/Task/PDFLetterCommon.php [hunk 1]
```diff
@@ -228,6 +228,8 @@ private static function resolveTokens($html_message, $contact, $contribution, $m
       // no change to normal behaviour to avoid risk of breakage
       $tokenHtml = CRM_Utils_Token::replaceContributionTokens($tokenHtml, $contribution, TRUE, $messageToken);
     }
+    $domain = CRM_Core_BAO_Domain::getDomain();
+    $tokenHtml = CRM_Utils_Token::replaceDomainTokens($tokenHtml, $domain, TRUE, $messageToken);
```
Comment (mattwire) [inline CRM/Contribute/Form/Task/PDFLetterCommon.php:232]: @seamuslee001 Can you put this above `replaceContactTokens()` - we found a scenario with the mailings recently when for some reason `replaceContactTokens()` stripped the domain tokens and they didn't get set.

**Review by mattwire — CHANGES_REQUESTED** [review]
@seamuslee001 Just one request to move a line of code and then this is good to merge.

**Comment by seamuslee001** [comment]
@mattwire done now

**Comment by monishdeb** [comment]
Merging as per MOP tag.

**Pull request merged**